spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/test/AbstractSpannerIntegrationTest.java

```diff
@@ -100,10 +99,7 @@
 
 	@BeforeClass
 	public static void checkToRun() {
-		assumeThat(
-				"Spanner integration tests are disabled. Please use '-Dit.spanner=true' "
-						+ "to enable them. ",
-				System.getProperty("it.spanner"), is("true"));
+		assumeThat(System.getProperty("it.spanner")).isEqualTo("true");
```

Comment:
No way to pass the error message?

Reply:
Hmm, it is possible to pass error message with AssertJ's assertThat but not with AssertJ's assumeThat.

I changed it based on artem's comment: https://github.com/spring-cloud/spring-cloud-gcp/pull/1362#discussion_r245724531

Do we want to stick with the style before (which allows messages) or move to assertJ?

Reply:
Why doesn't `.as()` work for you here?
```
Assumptions.assumeThat("foo")
		.as("Foo assumption")
		.isEqualTo("bar")
```

This is fully equivalent to:
```
Assume.assumeThat("Foo assumption", "foo", is("bar"));
```

Reply:
I tried using .as(...), however, it does not print the message when the assumption is broken. It seems to only print the message if you use as(...) with an assertThat(...).

Reply:
We talk here about `assume`. You can't compare `assertThat()` with `assumeThat()`.

This is an output of the ignored test for the mentioned assumption:
```
org.junit.AssumptionViolatedException: assumption was not met due to: [Foo assumption] 
Expecting:
 <"foo">
to be equal to:
 <"bar">
but was not.
```

Reply:
Ah I see, I will add it with the `.as(...)` then.

Just to clarify - where does one see the ignored output for the test like that? For me, I tested by adding the `assumeThat(...).as(...)` and then I run `mvn test` in the directory. I end up just seeing that the test is skipped; however I didn't see any message so I originally thought `.as(...)` was not doing anything.

Reply:
You should see that in the test report. Isn't it?

Reply:
Is test report the stuff that gets printed out on the console?

I don't see any message even after adding it; this is my console output after adding to Datastore sample test:

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running com.example.ApplicationTests
11:13:08.038 [main] DEBUG org.springframework.test.context.junit4.SpringJUnit4ClassRunner - SpringJUnit4ClassRunner constructor called with [class com.example.ApplicationTests]
11:13:08.042 [main] DEBUG org.springframework.test.context.BootstrapUtils - Instantiating CacheAwareContextLoaderDelegate from class [org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate]
11:13:08.049 [main] DEBUG org.springframework.test.context.BootstrapUtils - Instantiating BootstrapContext using constructor [public org.springframework.test.context.support.DefaultBootstrapContext(java.lang.Class,org.springframework.test.context.CacheAwareContextLoaderDelegate)]
11:13:08.065 [main] DEBUG org.springframework.test.context.BootstrapUtils - Instantiating TestContextBootstrapper for test class [com.example.ApplicationTests] from class [org.springframework.boot.test.context.SpringBootTestContextBootstrapper]
11:13:08.077 [main] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Neither @ContextConfiguration nor @ContextHierarchy found for test class [com.example.ApplicationTests], using SpringBootContextLoader
11:13:08.080 [main] DEBUG org.springframework.test.context.support.AbstractContextLoader - Did not detect default resource location for test class [com.example.ApplicationTests]: class path resource [com/example/ApplicationTests-context.xml] does not exist
11:13:08.080 [main] DEBUG org.springframework.test.context.support.AbstractContextLoader - Did not detect default resource location for test class [com.example.ApplicationTests]: class path resource [com/example/ApplicationTestsContext.groovy] does not exist
11:13:08.080 [main] INFO org.springframework.test.context.support.AbstractContextLoader - Could not detect default resource locations for test class [com.example.ApplicationTests]: no resource found for suffixes {-context.xml, Context.groovy}.
11:13:08.121 [main] DEBUG org.springframework.test.context.support.ActiveProfilesUtils - Could not find an 'annotation declaring class' for annotation type [org.springframework.test.context.ActiveProfiles] and class [com.example.ApplicationTests]
11:13:08.255 [main] DEBUG org.springframework.boot.test.context.SpringBootTestContextBootstrapper - @TestExecutionListeners is not present for class [com.example.ApplicationTests]: using defaults.
11:13:08.255 [main] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Loaded default TestExecutionListener class names from location [META-INF/spring.factories]: [org.springframework.boot.test.mock.mockito.MockitoTestExecutionListener, org.springframework.boot.test.mock.mockito.ResetMocksTestExecutionListener, org.springframework.boot.test.autoconfigure.restdocs.RestDocsTestExecutionListener, org.springframework.boot.test.autoconfigure.web.client.MockRestServiceServerResetTestExecutionListener, org.springframework.boot.test.autoconfigure.web.servlet.MockMvcPrintOnlyOnFailureTestExecutionListener, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverTestExecutionListener, org.springframework.test.context.web.ServletTestExecutionListener, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener, org.springframework.test.context.support.DependencyInjectionTestExecutionListener, org.springframework.test.context.support.DirtiesContextTestExecutionListener, org.springframework.test.context.transaction.TransactionalTestExecutionListener, org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener]
11:13:08.270 [main] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Using TestExecutionListeners: [org.springframework.test.context.web.ServletTestExecutionListener@a1153bc, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener@1aafa419, org.springframework.boot.test.mock.mockito.MockitoTestExecutionListener@47eaca72, org.springframework.boot.test.autoconfigure.SpringBootDependencyInjectionTestExecutionListener@55141def, org.springframework.test.context.support.DirtiesContextTestExecutionListener@55182842, org.springframework.test.context.transaction.TransactionalTestExecutionListener@3270d194, org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener@235834f2, org.springframework.boot.test.mock.mockito.ResetMocksTestExecutionListener@5656be13, org.springframework.boot.test.autoconfigure.restdocs.RestDocsTestExecutionListener@4218d6a3, org.springframework.boot.test.autoconfigure.web.client.MockRestServiceServerResetTestExecutionListener@76505305, org.springframework.boot.test.autoconfigure.web.servlet.MockMvcPrintOnlyOnFailureTestExecutionListener@14cd1699, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverTestExecutionListener@77888435]
11:13:08.271 [main] DEBUG org.springframework.test.annotation.ProfileValueUtils - Retrieved @ProfileValueSourceConfiguration [null] for test class [com.example.ApplicationTests]
11:13:08.272 [main] DEBUG org.springframework.test.annotation.ProfileValueUtils - Retrieved ProfileValueSource type [class org.springframework.test.annotation.SystemProfileValueSource] for class [com.example.ApplicationTests]
11:13:08.273 [main] DEBUG org.springframework.test.annotation.ProfileValueUtils - Retrieved @ProfileValueSourceConfiguration [null] for test class [com.example.ApplicationTests]
11:13:08.273 [main] DEBUG org.springframework.test.annotation.ProfileValueUtils - Retrieved ProfileValueSource type [class org.springframework.test.annotation.SystemProfileValueSource] for class [com.example.ApplicationTests]
11:13:08.273 [main] DEBUG org.springframework.test.annotation.ProfileValueUtils - Retrieved @ProfileValueSourceConfiguration [null] for test class [com.example.ApplicationTests]
11:13:08.273 [main] DEBUG org.springframework.test.annotation.ProfileValueUtils - Retrieved ProfileValueSource type [class org.springframework.test.annotation.SystemProfileValueSource] for class [com.example.ApplicationTests]
11:13:08.277 [main] DEBUG org.springframework.test.context.support.AbstractDirtiesContextTestExecutionListener - Before test class: context [DefaultTestContext@5b7a5baa testClass = ApplicationTests, testInstance = [null], testMethod = [null], testException = [null], mergedContextConfiguration = [WebMergedContextConfiguration@776aec5c testClass = ApplicationTests, locations = '{}', classes = '{class com.example.DatastoreRepositoryExample, class com.example.DatastoreRepositoryExample}', contextInitializerClasses = '[]', activeProfiles = '{}', propertySourceLocations = '{classpath:application-test.properties}', propertySourceProperties = '{org.springframework.boot.test.context.SpringBootTestContextBootstrapper=true, server.port=0}', contextCustomizers = set[org.springframework.boot.test.context.filter.ExcludeFilterContextCustomizer@1fc2b765, org.springframework.boot.test.json.DuplicateJsonObjectContextCustomizerFactory$DuplicateJsonObjectContextCustomizer@289d1c02, org.springframework.boot.test.mock.mockito.MockitoContextCustomizer@0, org.springframework.boot.test.web.client.TestRestTemplateContextCustomizer@6eceb130, org.springframework.boot.test.autoconfigure.properties.PropertyMappingContextCustomizer@0, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverContextCustomizerFactory$Customizer@31f924f5], resourceBasePath = 'src/main/webapp', contextLoader = 'org.springframework.boot.test.context.SpringBootContextLoader', parent = [null]], attributes = map['org.springframework.test.context.web.ServletTestExecutionListener.activateListener' -> false]], class annotated with @DirtiesContext [false] with mode [null].
11:13:08.891 [main] DEBUG org.springframework.test.context.support.AbstractDirtiesContextTestExecutionListener - After test class: context [DefaultTestContext@5b7a5baa testClass = ApplicationTests, testInstance = [null], testMethod = [null], testException = [null], mergedContextConfiguration = [WebMergedContextConfiguration@776aec5c testClass = ApplicationTests, locations = '{}', classes = '{class com.example.DatastoreRepositoryExample, class com.example.DatastoreRepositoryExample}', contextInitializerClasses = '[]', activeProfiles = '{}', propertySourceLocations = '{classpath:application-test.properties}', propertySourceProperties = '{org.springframework.boot.test.context.SpringBootTestContextBootstrapper=true, server.port=0}', contextCustomizers = set[org.springframework.boot.test.context.filter.ExcludeFilterContextCustomizer@1fc2b765, org.springframework.boot.test.json.DuplicateJsonObjectContextCustomizerFactory$DuplicateJsonObjectContextCustomizer@289d1c02, org.springframework.boot.test.mock.mockito.MockitoContextCustomizer@0, org.springframework.boot.test.web.client.TestRestTemplateContextCustomizer@6eceb130, org.springframework.boot.test.autoconfigure.properties.PropertyMappingContextCustomizer@0, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverContextCustomizerFactory$Customizer@31f924f5], resourceBasePath = 'src/main/webapp', contextLoader = 'org.springframework.boot.test.context.SpringBootContextLoader', parent = [null]], attributes = map['org.springframework.test.context.web.ServletTestExecutionListener.activateListener' -> false]], class annotated with @DirtiesContext [false] with mode [null].
[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.976 s - in com.example.ApplicationTests
[INFO] 
[INFO] Results:
[INFO] 
[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 5.469 s
[INFO] Finished at: 2019-01-23T11:13:09-05:00
[INFO] Final Memory: 32M/487M
[INFO] ------------------------------------------------------------------------
```

Reply:
OK. Is it printed when you use `assumeThat()` from JUnit?

Reply:
This is a difference in runner -- IDE's default runner prints failed assumption messages, but the default junit runner does not, so "mvn test" does not surface the output anywhere.

http://junit.sourceforge.net/javadoc/org/junit/Assume.html

> A failed assumption does not mean the code is broken, but that the test provides no useful information. The default JUnit runner treats tests with failing assumptions as ignored.

Reply:
Ah you're right. So running the test in the IDE will produce the error messages but not within the terminal; i'll be changing it to use `.as(...)` then.